mathics/core/rules.py

```diff
@@ -254,6 +254,13 @@ def apply_rule(
         new = self.replace.replace_vars(vars)
         new.options = options
 
+        while new.has_form("System`Condition", 2):
+            new, cond = new.get_elements()
+            if isinstance(cond, Expression):
+                cond = cond.evaluate(evaluation)
+            if cond is not SymbolTrue:
```

Comment:
It would be great if we could move all of this into Condition evaluation. 

What would it take to _not_ have to add `if cond is not SymbolTrue`?

Reply:
Line 260 looks like an evaluation. And why is it done only if it is an Expression? 

But, Ok, I'll wait until this is finished.